docs/cli-tool/kustomize-schema-generator.md

```diff
@@ -0,0 +1,170 @@
+---
+title: Kustomize Schema Generator
+---
+
+
+[kruise-api](https://github.com/openkruise/kruise-api) provides tools for openkruise to generate `kustomize schema` files that support openkruise CRD, and provides the current available `OpenAPI schema`.
+
+## Schema Generator
+
+When using `kustomize` to manage applic ations, in order to use `strategic merge patches (SMPs)` to process environment variables or any array type fields in resource definitions when using openkruise resources, the `OpenAPI schema` of openkruise CRD needs to contain the specified `patch strategy`. `Schema Generator` supports the function of quickly generating `kustomize schema` files based on the resource definition of openkruise. The specific way of generating `kustomize schema` files can be viewed in [README.md](https://github.com/openkruise/kruise-api/blob/master/cmd/gen-schema/README.md).
+
+## Schema Usage
+
+### How to specify the schema file
+
+[kruise-api](https://github.com/openkruise/kruise-api) provides available `OpenAPI schema` files, you can use the latest version or specify the version of the `OpenAPI schema` file, just add the configuration block in `kustomization.yaml` like:
+
+Use the latest version
+
+```yaml
+openapi:
+  path: https://raw.githubusercontent.com/openkruise/kruise-api/master/schema/openkruise_all_k8s_kustomize_schema.json
+```
+
+Use the specified version
+
+```yaml
+openapi:
+  path: https://raw.githubusercontent.com/openkruise/kruise-api/raw/<tag>/schema/argo_all_k8s_kustomize_schema.json
+```
+
+Or you can download [kruise-api](https://github.com/openkruise/kruise-api) to the local, and specify the local file path in the configuration block:
+
+```yaml
+openapi:
+  path: <kruise-api-local-path>/schema/openkruise_all_k8s_kustomize_schema.json
+```
+
+### Example
+
+Take `sidecarset` as an example to introduce how to use `kustomize OpenAPI schema` to support the merge strategy of openkruise CRD in kustomize.
+
+A complete definition of `sidecarset` is as follows:
```

Comment:
plz illustrate how to delete an entry in the array.

Reply:
Has illustrated